Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -59,6 +60,18 @@ - (void)loadView
     _contactsViewHelper.delegate = self;
     _nonContactAccountSet = [NSMutableSet set];
 
+    ReminderView *contactsPermissionReminderView = [[ReminderView alloc]
+        initWithText:NSLocalizedString(@"COMPOSE_SCREEN_MISSING_CONTACTS_PERMISSION",
+                         @"Multiline label explaining why compose-screen contact picker is empty.")
+           tapAction:^{
+               [[UIApplication sharedApplication] openSystemSettings];
+           }];
+    [self.view addSubview:contactsPermissionReminderView];
+    [contactsPermissionReminderView autoPinWidthToSuperview];
+    [contactsPermissionReminderView autoPinEdgeToSuperviewMargin:ALEdgeTop];
+    _hideContactsPermissionReminderViewConstraint =
+        [contactsPermissionReminderView autoSetDimension:ALDimensionHeight toSize:0];
```

Comment:
Well, if we show the search bar (see other comment), you might have:

* This header.
* Search bar.
* "find by phone number".
* "no contacts"?  (I think you suppress that when this header is visible?)
* two different possible "invite by sms" cells.